src/modules/content-files/content-files.c

```diff
@@ -0,0 +1,345 @@
+        flux_log_error (h, "error responding to kvs-checkpoint.get request");
+    free (data);
+    return;
+error:
+    if (flux_respond_error (h, msg, errno, errstr) < 0)
+        flux_log_error (h, "error responding to kvs-checkpoint.get request");
+    free (data);
+}
+
+/* Handle a kvs-checkpoint.put request from the rank 0 kvs module.
+ * The KVS stores its last root reference here for restart purposes.
+ */
+void checkpoint_put_cb (flux_t *h,
+                        flux_msg_handler_t *mh,
+                        const flux_msg_t *msg,
+                        void *arg)
+{
+    struct content_files *ctx = arg;
+    const char *key;
+    const char *value;
+    const char *errstr = NULL;
+
+    if (flux_request_unpack (msg,
+                             NULL,
+                             "{s:s s:s}",
+                             "key",
+                             &key,
+                             "value",
+                             &value) < 0)
+        goto error;
+    if (filedb_put (ctx->dbpath, key, value, strlen (value), &errstr) < 0)
+        goto error;
+    if (flux_respond (h, msg, NULL) < 0)
+        flux_log_error (h, "error responding to kvs-checkpoint.put request");
+    return;
+error:
+    if (flux_respond_error (h, msg, errno, errstr) < 0)
+        flux_log_error (h, "error responding to kvs-checkpoint.put request");
+}
+
+/* Destroy module context.
+ */
+static void content_files_destroy (struct content_files *ctx)
+{
+    if (ctx) {
+        int saved_errno = errno;
+        flux_msg_handler_delvec (ctx->handlers);
+        free (ctx->dbpath);
+        free (ctx);
+        errno = saved_errno;
+    }
+}
+
+/* Table of message handler callbacks registered below.
+ * The topic strings in the table consist of <service name>.<method>.
+ */
+static const struct flux_msg_handler_spec htab[] = {
+    { FLUX_MSGTYPE_REQUEST, "content-backing.load",    load_cb, 0 },
+    { FLUX_MSGTYPE_REQUEST, "content-backing.store",   store_cb, 0 },
+    { FLUX_MSGTYPE_REQUEST, "kvs-checkpoint.get", checkpoint_get_cb, 0 },
+    { FLUX_MSGTYPE_REQUEST, "kvs-checkpoint.put", checkpoint_put_cb, 0 },
+    FLUX_MSGHANDLER_TABLE_END,
+};
+
+/* Create module context and perform some initialization.
+ */
+static struct content_files *content_files_create (flux_t *h)
+{
+    struct content_files *ctx;
+    const char *backing_path;
+    const char *tmp;
+
+    if (!(ctx = calloc (1, sizeof (*ctx))))
+        return NULL;
+    ctx->h = h;
+
+    /* Some tunables:
+     * - the hash function, e.g. sha1, sha256
+     * - the maximum blob size
+     * - path to sqlite file
+     */
+    if (!(ctx->hashfun = flux_attr_get (h, "content.hash"))) {
+        flux_log_error (h, "content.hash");
+        goto error;
+    }
+    if (!(tmp = flux_attr_get (h, "content.blob-size-limit"))) {
+        flux_log_error (h, "content.blob-size-limit");
+        goto error;
+    }
+    ctx->blob_size_limit = strtoul (tmp, NULL, 10);
```

Comment:
should check for error on blob-size-limit input?  Noticed not done in `content-sqlite`.  perhaps test for invalid blob-size-limit good idea too?